sechub-wrapper-prepare/src/main/java/com/mercedesbenz/sechub/wrapper/prepare/modules/JGitAdapter.java

```diff
@@ -0,0 +1,93 @@
+package com.mercedesbenz.sechub.wrapper.prepare.modules;
+
+import static com.mercedesbenz.sechub.wrapper.prepare.cli.PrepareWrapperEnvironmentVariables.PDS_PREPARE_CREDENTIAL_PASSWORD;
+import static com.mercedesbenz.sechub.wrapper.prepare.cli.PrepareWrapperEnvironmentVariables.PDS_PREPARE_CREDENTIAL_USERNAME;
+
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+
+import javax.crypto.SealedObject;
+
+import org.eclipse.jgit.api.CloneCommand;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.transport.UsernamePasswordCredentialsProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.mercedesbenz.sechub.commons.core.security.CryptoAccess;
+
+@Component
+public class JGitAdapter {
+
+    private static final Logger LOG = LoggerFactory.getLogger(JGitAdapter.class);
+
+    public void clone(GitContext gitContext) {
+        String location = transformLocationToURL(gitContext.getLocation());
+        Map<String, SealedObject> credentialMap = gitContext.getCredentialMap();
+
+        String username = getUserNameFromMap(credentialMap);
+        String password = getPasswordFromMap(credentialMap);
+
+        CloneCommand command = Git.cloneRepository().setURI(location).setDirectory(Paths.get(gitContext.getUploadDirectory()).toFile());
+
+        if (!(username == null || password == null)) {
```

Comment:
This proposal means the same as your conditional statement but might be easier to read.
```suggestion
        if (username != null && password != null) {
```